drivers/usbdev_synopsys_dwc2/usbdev_synopsys_dwc2.c

```diff
@@ -1035,7 +1050,17 @@ static int _usbdev_get(usbdev_t *dev, usbopt_t opt,
             break;
         case USBOPT_MAX_SPEED:
             assert(max_len == sizeof(usb_speed_t));
-            *(usb_speed_t *)value = USB_SPEED_FULL;
+            if (IS_USED(DWC2_USB_OTG_HS_ENABLED)) {
+                *(usb_speed_t *)value = USB_SPEED_HIGH;
```

Comment:
This should check the actual speed of the device, not just claim high speed if *a* high speed peripheral is on the MCU.

Reply:
And fixed with the latest push